### PR TITLE
doc/developer: improve release documentation

### PR DESCRIPTION
**Pull request by benesch**

### CODE REVIEW

File: doc/developer/release-checklist.md [hunk 1]
```diff
@@ -1,30 +1,99 @@
...
-- [ ] load/soak tests run on -rc tagged commit
-  - command should be:
-    ```
-    dc.sh clean-load-test
+- [ ] Choose the name for the release candidate following the format
+  `v<VERSION>-rc<N>`, where _N_ starts at 1. For example, the first RC for
+  v0.2.3 would be called v0.2.3-rc1.
+
+- [ ] Update the version field in [`src/materialized/Cargo.toml`](../../src/materialized/Cargo.toml)
+      and commit that change.
```
Comment (quodlibetor) [inline doc/developer/release-checklist.md]: maybe add an "e.g. `version = "0.2.3-rc1"`" to make it super clear that the -rc` shows up in the cargo.toml?

File: doc/developer/release-checklist.md [hunk 2]
```diff
@@ -1,30 +1,99 @@
...
+
+- [ ] Check out the final RC tag.
+
+- [ ] Update the version field in [`src/materialized/Cargo.toml`](../../src/materialized/Cargo.toml)
+      and commit that change.
+
+- [ ] Create the release tag on that commit.
+
+  ```shell
+  tag=v<VERSION>-rc<N>
```
Comment (quodlibetor) [inline doc/developer/release-checklist.md]: no `-rc` here.

Comment (benesch) [reply doc/developer/release-checklist.md]: oops, thanks

File: doc/developer/release-checklist.md (hunk 1 above)
Comment (benesch) [reply doc/developer/release-checklist.md]: done

**Pull request merged**